### PR TITLE
Override signMessage for SmartWallet class

### PR DESCRIPTION
**Pull request by nkrishang**
## Problem solved

Uses EIP-712 signature in `signMessage` for the updated EIP-1271 signature verification in smart wallet contracts (https://github.com/thirdweb-dev/contracts/pull/572)

## Changes made

- [ ] Public API changes: list the public API changes made if any
- [x] Internal API changes: explain the internal logic changes
    - **Before:** `SmartWallet.signMessage` directly uses ethers-js `signer.signMessage` to sign the message hash of the data to sign (i.e. `signMessage(ethers.utils.hashMessage(data))`)
    - **Now:** `SmartWallet.signMessage` signes EIP-712 typed data of the data to sign mixed with the smart wallet's domain separator. If this yields in invalid signature, the function falls back to the legacy method of signing.

## How to test

- [ ] Automated tests: link to unit test file
- [x] Manual tests: step by step instructions on how to test
    - https://linear.app/thirdweb/issue/PLAT-1313/test-jam-[smartwallet][auth]-sign-in-with-smart-wallet

### CODE REVIEW

File: packages/wallets/src/evm/wallets/smart-wallet.ts [hunk 1]
```diff
@@ -76,6 +78,90 @@ export class SmartWallet extends AbstractClientWallet<
...
+    const isDeployed = await connector.isDeployed();
+    if (!isDeployed) {
+      console.log(
+        "Account contract not deployed yet. Deploying account before signing message",
+      );
+      const tx = await erc4337Signer.sendTransaction({
+        to: await this.getAddress(),
+        data: "0x",
+      });
+      await tx.wait();
```
Comment (joaquim-verges) [inline packages/wallets/src/evm/wallets/smart-wallet.ts:98]: you can just use `connector.deployIfNeeded()` here

File: packages/wallets/src/evm/wallets/smart-wallet.ts [hunk 2]
```diff
@@ -76,6 +78,90 @@ export class SmartWallet extends AbstractClientWallet<
...
+    const AccountMessage = [{ name: "message", type: "bytes" }];
+
+    /**
+     * We first try to sign the EIP-712 typed data i.e. the message mixed with the smart wallet's domain separator.
+     * If this fails, we fallback to the legacy signing method.
+     */
+    try {
+      const signature = await (
+        signer as providers.JsonRpcSigner
+      )._signTypedData(
```
Comment (joaquim-verges) [inline packages/wallets/src/evm/wallets/smart-wallet.ts:118]: can we use `signTypedDataInternal()` instead ? already handles a lot of this for you and its better to centralize this logic in one place

File: packages/wallets/src/evm/wallets/smart-wallet.ts [hunk 3]
```diff
@@ -76,6 +78,90 @@ export class SmartWallet extends AbstractClientWallet<
...
+            [utils.hashMessage(message)],
+          ),
+        },
+      );
+
+      if (!signature) {
+        throw new Error("Failed to sign message");
+      }
+
+      const isValid = await checkContractWalletSignature(
```
Comment (joaquim-verges) [inline packages/wallets/src/evm/wallets/smart-wallet.ts:138]: I don't think it makes sense to validate the signature here, will also slow down the process quite a bit

Comment (nkrishang) [reply packages/wallets/src/evm/wallets/smart-wallet.ts:138]: I don't think there's another way to check whether to fallback to legacy method of signing. 

So this seems necessary to maintain backward compatibility with existing smart wallets out there.

File: packages/wallets/src/evm/wallets/smart-wallet.ts (hunk 2 above)
Comment (nkrishang) [reply packages/wallets/src/evm/wallets/smart-wallet.ts:118]: I tried importing it but that didn't work; I'll try again.

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**